File: jems-ui/src/app/project/model/ProjectPeriod.ts
export class ProjectPeriod {
  projectId: number;
  periodNumber: number;
  start: number;
  end: number;

  constructor(projectId: number, periodNumber: number, start: number, end: number) {
    this.projectId = projectId;
    this.periodNumber = periodNumber;
    this.start = start;
    this.end = end;
  }
}

File: jems-server/src/main/kotlin/io/cloudflight/jems/server/project/service/model/ProjectPeriod.kt
package io.cloudflight.jems.server.project.service.model

import java.time.LocalDate

data class ProjectPeriod (
    override val number: Int,
    override val start: Int,
    override val end: Int,
    override var startDate: LocalDate? = null,
    override var endDate: LocalDate? = null,
) : ProjectPeriodBase
